google/cloud/internal/timer_queue_test.cc

```diff
@@ -81,6 +81,23 @@ TEST(TimerQueueTest, ScheduleEarlierTimerSingleRunner) {
   EXPECT_GT(*later_expire_time, *earlier_expire_time);
 }
 
+TEST(TimerQueueTest, ShutdownWithPendingTimersCancelledSingleRunner) {
+  TimerQueue tq;
+  auto const duration = std::chrono::seconds(60);
+  auto now = std::chrono::system_clock::now();
+  std::thread t([&tq] { tq.Service(); });
+  auto f = tq.Schedule(now + duration);
+  auto f2 = tq.Schedule(now + duration);
+
+  tq.Shutdown();
+  t.join();
+
+  auto expire_time = f.get();
+  EXPECT_THAT(expire_time, StatusIs(StatusCode::kCancelled));
```

Comment:
These are still timing dependent, albeit with admitted low probability, but I don't have a better solution short of checking that they are either cancelled or yield a time in the past.